tests/foreman/api/test_classparameters.py

```diff
@@ -0,0 +1,1035 @@
+        2.  Create a matcher for Parameter for some attribute.
+        3.  Set some Value for matcher.
+        4.  Set 'required' to true.
+
+        @assert: Error not raised for matcher value.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_negative_validate_default_value_with_regex(self):
+        """Error raised for default value not matching with regex.
+
+        @feature: Class Parameters - Optional Validation
+
+        @steps:
+
+        1.  Set override to True.
+        2.  Set default value that doesn't matches the regex of step 3.
+        3.  Validate this value with regex validator type and rule.
+
+        @assert: Error raised for default value not matching with regex.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_validate_default_value_with_regex(self):
+        """Error not raised for default value matching with regex.
+
+        @feature: Class Parameters - Optional Validation
+
+        @steps:
+
+        1.  Set override to True.
+        2.  Set default value that matches the regex of step 3.
+        3.  Validate this value with regex validator type and rule.
+
+        @assert: Error not raised for default value matching with regex.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_negative_validate_matcher_value_with_regex(self):
+        """Error raised for matcher value not matching with regex.
+
+        @feature: Class Parameters - Optional Validation
+
+        @steps:
+
+        1.  Set override to True.
+        2.  Create a matcher with value that doesn't matches
+        the regex of step 3.
+        3.  Validate this value with regex validator type and rule.
+
+        @assert: Error raised for matcher value not matching with regex.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_validate_matcher_value_with_regex(self):
+        """Error not raised for matcher value matching with regex.
+
+        @feature: Class Parameters - Optional Validation
+
+        @steps:
+
+        1.  Set override to True.
+        2.  Create a matcher with value that matches the regex of step 3.
+        3.  Validate this value with regex validator type and rule.
+
+        @assert: Error not raised for matcher value matching with regex.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_negative_validate_default_value_with_list(self):
+        """Error raised for default value not in list.
+
+        @feature: Class Parameters - Optional Validation
+
+        @steps:
+
+        1.  Set override to True.
+        2.  Set default value that doesn't matches the list of step 3.
+        3.  Validate this value with list validator type and rule.
+
+        @assert: Error raised for default value not in list.
```

Comment:
`Error is raised for default value that not in the list.` Please, correct all occurences